File: src/Estruturas_e_Bilbiotecas/1068.c
#include <stdio.h>
#include <stdlib.h>
#include <locale.h>
#include <math.h>
#include <string.h>
#include <stdbool.h>
int veri(char c[900000])
{
      bool tick = false;
      bool t = true;
      int u = 0;
      bool verifica = false;
      int i;
      int len = strlen(c);
      for (i = 0; i < len; i++)
      {
            if (c[i] == '(')
            {
                  u = u + 1;
            }
            if (c[i] == ')')
            {
                  u = u - 1;
            }
            if (u < 0)
            {
                  return 1;
            }
      }
      if (u == 0)
      {
            return 0;
      }
      if (u != 0)
      {
            return 1;
      }
}
int main()
{
      setlocale(LC_ALL, "");
      bool verifica = false;
      int h = 0;
      char entrada[900000];

      while (scanf("%s", &entrada) != EOF)
      {
            int i, len = strlen(entrada);
            int r = veri(entrada);
            switch (r)
            {
            case 1:
                  printf("incorrect\n");
                  break;
            case 0:
                  printf("correct\n");
                  break;
            default:
                  break;
            }
      }
      return 0;
}

File: src/Estruturas_e_Bilbiotecas/1068.java
package Estruturas_e_Bilbiotecas;

import java.io.IOException;
import java.util.Scanner;

public class Main {
    static int veri(String c) {
        boolean tick = false;
        boolean t = true;
        int u = 0;
        boolean verifica = false;
        int i;
        int len = c.length();
        for (i = 0; i < len; i++) {
            if (c.charAt(i) == '(') {
                u = u + 1;
            }
           
            if (c.charAt(i) == ')') {
                u = u - 1;
            }
            if (u < 0) {
                return 1;
            }
        }
        if (u == 0) {
            return 0;
        }
        if (u != 0) {
            return 1;
        }
        return len;
    }

    public static void main(String[] args) throws IOException {
        Scanner input = new Scanner(System.in);
        boolean verifica = false;
        int h = 0;

        String entrada;
        while (input.hasNext()) {
            entrada = input.nextLine();

            int r = veri(entrada);
            switch (r) {
                case 1:
                    System.out.println("incorrect");

                    break;
                case 0:

                    System.out.println("correct");
                    break;
                default:
                    break;
            }
        }

    }
}

